diesel/src/associations/belongs_to.rs

```diff
@@ -27,8 +27,9 @@ impl<Parent, Child, Iter> GroupedBy<Parent> for Iter where
         let id_indices: HashMap<_, _> = parents.iter().enumerate().map(|(i, u)| (u.id(), i)).collect();
         let mut result = parents.iter().map(|_| Vec::new()).collect::<Vec<_>>();
         for child in self {
-            let index = id_indices[child.foreign_key()];
-            result[index].push(child);
+            if let Some(index) = child.foreign_key().map(|i| id_indices[i]) {
+                result[index].push(child);
+            }
         }
```

Comment:
I think there is a way to write that a little cleaner/functional, with flat_map and collect/fold. I'm just reading through this quickly though, and maybe come back later with an actual refactor suggestion.